src/service/metrics/prom.rs

```diff
@@ -313,19 +311,12 @@ pub async fn remote_write(
     }
 
     for (stream_name, stream_data) in metric_data_map {
+        // stream_data could be empty if metric value is nan, check it
+        if stream_data.is_empty() {
+            continue;
+        }
         // write to file
         let mut stream_file_name = "".to_string();
-
-        // check if we are allowed to ingest
-        if db::compact::retention::is_deleting_stream(
-            org_id,
-            &stream_name,
-            StreamType::Metrics,
-            None,
-        ) {
-            return Err(anyhow::anyhow!("stream [{stream_name}] is being deleted"));
-        }
-
```

Comment:
You can't delete this section, you can report error and continue.

Reply:
> You can't delete this section, you can report error and continue.

This is the code to delete this time, I put the check in advance.

Reply:
i saw that, you check it for each event, earlier we only need check once, do you think checking for every event has better performance?

Reply:
> i saw that, you check it for each event, earlier we only need check once, do you think checking for every event has better performance?

I had noticed that each metric_name could be checked multiple times. Since `is_deleting_stream` is implemented using `CACHE.contains`, and `Cache` is implemented as a Set struct, the `check_deleting` operation is much cheaper than further operations involved.